### PR TITLE
Implement `{when_iso_local}` placeholder

### LINKED ISSUE #2362: Add `{when_iso_local}` placeholder
### 1. What version of Caddy are you using (`caddy -version`)?

Git commit 8f7a1d6a2 from Nov 18, 2018.

### 2. What are you trying to do?

I prefer the ISO 8601 timestamp format but when perusing the log file it is friendlier if the human readable timestamps are in the local timezone of the server rather than UTC. So I'd like to propose adding a `{when_iso_local}` placeholder. The implementation is trivial:

```
diff --git a/caddyhttp/httpserver/replacer.go b/caddyhttp/httpserver/replacer.go
index 945b93e..8755f40 100644
--- a/caddyhttp/httpserver/replacer.go
+++ b/caddyhttp/httpserver/replacer.go
@@ -368,6 +368,8 @@ func (r *replacer) getSubstitution(key string) string {
                return url.QueryEscape(r.request.URL.RequestURI())
        case "{when}":
                return now().Format(timeFormat)
+       case "{when_iso_local}":
+               return now().Format(timeFormatISO)
        case "{when_iso}":
                return now().UTC().Format(timeFormatISOUTC)
        case "{when_unix}":
@@ -539,6 +541,7 @@ func (r *replacer) Set(key, value string) {

 const (
        timeFormat        = "02/Jan/2006:15:04:05 -0700"
+       timeFormatISO     = "2006-01-02T15:04:05" // ISO 8601 with timezone to be assumed as local
        timeFormatISOUTC  = "2006-01-02T15:04:05Z" // ISO 8601 with timezone to be assumed as UTC
        headerContentType = "Content-Type"
        contentTypeJSON   = "application/json"
```

For automated log file analysis I also include the `{when_unix_ms}` placeholder. This gives analysis tools an unambiguous timestamp while giving humans an easier to interpret string via the `{when_iso_local}` placeholder. Yes, this is itself problematic if you have personnel in different time zones who need to peruse the logs. But in that case you can either use the UTC timestamps and/or a tool to convert the UTC timestamps to each person's local timezone. This enhancement is primarily for people who are using Caddy as a personal web server.

### PR DESCRIPTION
**Pull request by krader1961**
This implements the `{when_iso_local}` placeholder. This is like the
`{when_iso}` placeholder but the output is in the current timezone
rather than UTC.

Resolves #2362

### CODE REVIEW

**Comment by krader1961** [comment]
The comments I added about the use of `FixedTimezone()` is because I was surprised the value was in seconds versus hours. So my first attempt at adding a test case failed. Yes, an experienced Go programmer would know that but it seems to me to be sufficiently obscure to warrant a comment.

**Comment by krader1961** [comment]
Why is the build failing with the errors below since my change does not change any imports and `go fmt` on my system doesn't modify my change?

```
gometalinter --disable-all -E vet -E gofmt -E misspell -E ineffassign -E goimports -E deadcode --tests --vendor ./...
caddyhttp\httpserver\replacer.go:1::warning: file is not gofmted with -s (gofmt)
caddyhttp\httpserver\replacer.go:1::warning: file is not goimported (goimports)
```

**Comment by tobya** [comment]
> Also, I would have updated the documentation at https://caddyserver.com/docs/placeholders but it wasn't obvious how to do so.

The docs repo is looked after by @mholt. If you put here the changes that need to be added to docs Matt will add them to the repo.

**Comment by krader1961** [comment]
> If you put here the changes that need to be added to docs Matt will add them to the repo.

Okay. In this case it's pretty simple. Modify https://caddyserver.com/docs/placeholders by adding 

```
{when_iso_local} | Timestamp in the format 2006-01-02T15:04:05 in local TZ
```

**Comment by tobya** [comment]
If you run 
   
    gofmt -s -w replacer.go 

on the two files and submit any changes that should do it.

**Comment by krader1961** [comment]
> gofmt -s -w replacer.go

Done

File: caddy/main.go [hunk 1]
```diff
@@ -20,6 +20,8 @@
 package main
 
 import "github.com/mholt/caddy/caddy/caddymain"
+import _ "github.com/krader1961/caddy-geoip"
+import _ "github.com/krader1961/ipfilter"
```
Comment (francislavoie) [inline caddy/main.go]: This seems wrong

Comment (krader1961) [reply caddy/main.go]: /facepalm /embarrassment Yes, my custom config commit should not have been included. I've force pushed the proper change.

**Comment by tobya** [comment]
@francislavoie Are you happy to approve?

**Comment by tobya** [comment]
Thank you  @krader1961 .  Sorry took so long to merge.

**Pull request merged**